Bottom: 55418b4e140bb331907c4791d2db95903c5a72d5
Top:    ce75d7c015701cd552af4fba4e8b6a114586d6f7
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-25 10:17:59 -0500

Small fixes to improve idempotency

Generally, adding changed_when parameters or tweaking tasks to ensure changed
statuses are reported correctly and tasks only run when needed.


---
diff --git a/roles/bacula-storage/tasks/main.yml b/roles/bacula-storage/tasks/main.yml
index 01125af5..ba5ccb7b 100644
--- a/roles/bacula-storage/tasks/main.yml
+++ b/roles/bacula-storage/tasks/main.yml
@@ -19,10 +19,13 @@
     backup: yes
 
 - name: Check if baculatape selinux module exists
-  stat: path=/etc/selinux/targeted/modules/active/modules/baculatape.pp
+  command: semodule -l 
   register: baculatape_pp
+  changed_when: false
+
 - import_tasks: selinux-tape-policy.yml
-  when: baculatape_pp.stat.exists is defined and baculatape_pp.stat.exists == false
+  when: "'baculatape\t1.0' not in baculatape_pp.stdout_lines"
+
 - name: Make sure bacula user is in the tape group
   user: 
     name: bacula
@@ -30,6 +33,8 @@
     append: yes
 - name: Test Bacula Storage configuration
   command: /usr/sbin/bacula-sd -t -c {{ bacula_storage_config_dir }}/bacula-sd.conf
+  changed_when: false
+
 - name: Enable and start Bacula Storage
   service: 
     name: bacula-sd
